vercel/vercel
# docs(sveltekit): re-introduce speed insights

### What's in there?

A customer recently leverage an outdated sveltekit example of ours to enable speed insights on a SvelteKit application, and had some issue with the `VERCEL_ANALYTICS_ID` env variable.

While migrating from pre-SvelteKit 1 to latest SvelteKit it seems that we removed the webvitals handling.
This PR brings back the web vitals collection, which allows displaying speed insights once deployed and enabled on Vercel.

### How to test?

Run the full example on Vercel with Speed insights enabled:

1. Extract the content of `vercel/examples/sveltekit-1` into its own folder
2. Install dependencies `pnpm i`
3. Deploy to vercel `vc deploy`
4. On Vercel dashboard, browse to the "Speed Insights" tab, enable and hit the redeploy button
5. Visit your application and browse back-and-forth the about page
6. Back to Vercel dashboard, insights should appear eventually 

<img width="1585" alt="image" src="https://github.com/vercel/vercel/assets/186268/596af435-2f57-4ae4-a228-044bfcea502c">


### Notes to reviewers

The previous implementation was leveraging vite to read `VERCEL_ANALYTICS_ID` from the environment, and use a replacement plugin to inject into the bundled code, at build time.

This time we're leveraging Vercel-adapter, which makes [Vercel env variables available to the sveltekit application](https://kit.svelte.dev/docs/adapter-vercel#environment-variables), through Svelte native module.

Note: if you wonder why we're using `$env/dynamic/private` and not  `$env/dynamic/public`, this is because we can't  change the variable name to be prefixed with `PUBLIC_` (or anything user would configure with `config.kit.env.publicPrefix`). When kit would support a list of variable prefix, then we'll revisit this example.
